bin/run_tests.sh

```diff
@@ -43,7 +43,18 @@ trap "echo """"; cat ""$BUILD_DIR""/run_tests.results ""$BUILD_DIR""/run_tests.f
 
 echo "$(echo "$TESTS" | wc -l) tests."
 
-START_TIME_TOTAL=$(date +%s%N | cut -b1-13)
+
+# get nanoseconds, with workaround for macOS
+nanosec () {
+  if date --help | grep nanoseconds > /dev/null; then
+    date +%s%N | cut -b1-13
+  else
+    date +%s000000 | cut -b1-13
```

Comment:
I think this should be `date +%s000000000` since nanoseconds are 10⁻⁹ s.
Can we reduce the overhead on non-macOS by doing something like
```
  (date +s%N || date +s%000000000) | cut -b1-13
```
instead?

Reply:
Ugh, embarrassing conversion mistake... :-) 
We can not do the suggested since date+s%N does not error on macOS. It returns a time with a capital N at the end.